healpy/src/_sphtools.pyx

```diff
@@ -275,7 +275,15 @@ def map2alm(m, lmax = None, mmax = None, niter = 3, use_weights = False,
     if use_weights:
         if datapath is None:
             datapath = get_datapath()
-        c_datapath = datapath
+        # For Python3: datapath must be a str, bdatapath must be bytes
+        if type(datapath) is unicode :
+          bdatapath = datapath.encode('UTF-8')
```

Comment:
Please match the style of the surrounding code and use four spaces for indentation.